Loginclient

TijoloAzul opened the pull request:
This solves #8

TijoloAzul committed:
changed from simple.client to client.simple

TijoloAzul committed:
created login client stub

TijoloAzul committed:
java was updated to java 8 minimum

TijoloAzul committed:
stub of scanner protocol to the login client

TijoloAzul committed:
the login client already does everything the the simple client does
is still missing login, relogin and logout

there will be two commands more hash and info
that show the hash and some information

TijoloAzul committed:
Login client already tries to login

TijoloAzul committed:
created a server scanner protocol
it should be possible
to see help
get the list of clients
confirm login of clients
logout clients
disconnect clients
exit application

In future should be possible to see a list of client information

There are a lot of bugs... yet

TijoloAzul committed:
Merge branch 'develop' into loginclient

# Conflicts:
#	src/main/java/pt/dinis/main/ServerScanner.java

TijoloAzul committed:
messages to server continue "Display.display"
but all kind of events are now "Display.info"

dealer exit command sends a disconnect message to all clients before
exiting

TijoloAzul committed:
o cliente já interpreta as mensagens do servidor
o login funciona no cliente mas o servidor tem que o gerir manualmente
tinha-me esquecido de passar as mensagens para lowercase

TijoloAzul committed:
ignoring user.conf file
using a sample file instead

TijoloAzul committed:
remove user.conf file
please take user.sample.file and create a new user.conf

diogodinisf commented on src/main/java/pt/dinis/client/login/LoginClientCommunication.java:
```diff
@@ -0,0 +1,98 @@
+                String message = in.readLine();
+                logger.debug("Receiving message " + message);
+                if(message == null) {
+                    Display.alert("Error receiving from server: disconnect");
```
não é que goste particularmente desta implementação, mas acho que só vale a pena voltar a implicar se encontrar algo melhor. por enquanto vai assim

TijoloAzul replied:
eu sei que não gostas... mas eu não sei como fazer melhor.

diogodinisf commented on src/main/java/pt/dinis/client/login/LoginClientCommunicationProtocol.java:
```diff
@@ -0,0 +1,100 @@
+            this.word = word;
+        }
+
+        String getWord() {
```
isto não deveria de ter um modificador?

TijoloAzul replied:
Certíssimo e feito

diogodinisf commented on src/main/java/pt/dinis/client/login/LoginClient.java:
```diff
@@ -0,0 +1,126 @@
+
+    private static final String DEFAULT_IP = "localhost";
+    private static final Integer DEFAULT_PORT = 1500;
+
```
não faz sentido ter as propriedades montadas e estares a usar isto

TijoloAzul replied:
sem desculpas de não usar a tua cena nova... feito

diogodinisf replied:
não é que goste especialmente deste boneco, mas é para mostar que por mim este comentário já está arrumado

diogodinisf commented on src/main/java/pt/dinis/client/login/LoginClient.java:
```diff
@@ -0,0 +1,126 @@
+            hash = null;
+            return true;
+        }
+        Display.info("Logout");
```
isto não é dentro fo if?

TijoloAzul replied:
Não, só informa que estamos a deslogar, no caso de estar logado.
Achas que não?

diogodinisf replied:
então, só entras no if se tiveres logado certo? Se entrares no if não vais mostrar a mensagem de 'Logout'. Só vais mostrar a mensagem se a pessoa já ester logged out antes de chamar o método

TijoloAzul replied:
É o que sempre disse, a mensagem tem que estar dentro do if...

Corrigido

diogodinisf commented on src/main/java/pt/dinis/main/ServerScannerProtocol.java:
```diff
@@ -0,0 +1,166 @@
+        }
+
+        if(MessageType.LOGIN.getKeys().contains(word)) {
+            return login(ids, words.get(2));
```
se escrever 'login all hash' o que acontece?

diogodinisf commented on src/main/java/pt/dinis/main/ServerScannerProtocol.java:
```diff
@@ -0,0 +1,166 @@
+
+    private static boolean login(Collection<Integer>ids, String hash) {
+        if(ids.size() > 1) {
+            Display.alert("Cannot reply to more than one login demand");
```
ok, já vi o que acontece

diogodinisf commented on src/main/java/pt/dinis/client/login/LoginClientCommunicationProtocol.java:
```diff
@@ -0,0 +1,100 @@
+
+        List<String> words = splitMessage(message);
+
+        // If message is empty, then it tries to start, login or relogin
```
de certeza?

TijoloAzul replied:
corta e cola... corrigido

diogodinisf commented on src/main/java/pt/dinis/client/login/LoginClientScannerProtocol.java:
```diff
@@ -0,0 +1,234 @@
+        List<String> words = splitMessage(message);
+
+        // If message is empty, then it tries to start, login or relogin
+        if(words.isEmpty()) {
```
sinceramente, não acho isto correto, se o cliente se enganou e enviou uma coisa vazia sem querer? não é razão válida para este trabalho

TijoloAzul replied:
Eu percebo o teu problema, mas eu queria evitar estar sempre a escrever start e login :)
Se insistires eu apago... mas tens que insistir

Seja como for o loginclient serve apenas para testar

diogodinisf reviewed (commented):
responde ao review

TijoloAzul committed:
using properties file in the login client
minor corrections

TijoloAzul committed:
simpler logout logic on LoginClient:
if command arrives there it always logs out and displays a message
anyway, the scanner is blocking the command when it is not logged out

hash command is now prettier
